pkg/operator/ceph/cluster/osd/status.go

```diff
@@ -159,7 +173,30 @@ func (c *Cluster) completeOSDsForAllNodes(config *provisionConfig, configOSDs bo
 		orchestrationStatusKey, provisioningLabelKey,
 	)
 
-	originalNodes, remainingNodes, completed, statuses, err := c.checkNodesCompleted(selector, config, configOSDs)
+	// For OSDs that are definitely update operations, we want to be able to defer the update until
+	// after we have created new OSDs in order to speed up cluster scale-out. Allow us to defer
+	// updating by keeping a list of deferred configmaps and processing those after creates have
+	// been done. If OSD provisioning doesn't report a "preexisting" status, then no configmaps will
+	// be deferred, and OSD create/update ops will be intermixed in whatever order is processed.
+	deferredConfigMaps := []*v1.ConfigMap{}
+	defer func() {
```

Comment:
Does this need to be in a `defer` func? Seems like it could just be at the end of the method. If there were a panic we wouldn't actually want to process these. If the reconcile restarts, it will just rebuilding the list of osds from the configmaps that would still exist.

Reply:
It is a little more code churn to put it at the end vs in a defer, though this would be a little more clear. I actually didn't realize until now that defer is still called after a panic. I should change this.

Reply:
The nested `for` loops make the code changes here intimidating. It would be best to have the code to update the deferred maps at the end and have the loops set a return value and break from the loops, but I don't want to take the risk right now of introducing a logic bug related to getting a return value wrong. I'd rather not do this, but I think this code will have to be replaced anyway to support updating OSDs in parallel, so I'm not worried about the long-term maintenance costs for this solution.